manifests/init.pp

```diff
@@ -439,6 +441,14 @@
     notify  => Service['snmpd'],
   }
 
+  # Debian 9 use systemd
+  if ( $facts['os']['name'] == 'Debian' and versioncmp($facts['os']['release']['major'], '9') >= 0 ) {
+    systemd::dropin_file { 'snmpd.conf':
+      unit    => 'snmpd.service',
+      content => epp($template_snmpd_service_dropin),
+    } ~> Class['systemd::systemctl::daemon_reload'] ~> Service['snmpd']
```

Comment:
FYI, you can set `daemon_reload => 'eager'` for dropin_file and then you don't need to explicitly notify daemon_reload

Reply:
yes indeed. But code is working on production now without issues. Just got stuck with Rspec tests don't know why.